src/pyramid/security.py

```diff
@@ -363,3 +363,50 @@ def has_permission(self, permission, context=None):
             )  # should never happen
         principals = authn_policy.effective_principals(self)
         return authz_policy.permits(context, principals, permission)
+
+
+@implementer(ISecurityPolicy)
+class CompatibilitySecurityPolicy(object):
```

Comment:
A naming convention nit: `LegacySessionCSRFStoragePolicy` exists to make csrf storage work like it used to. Probably use `Legacy` instead of `Compatibility` here as well.